### PR TITLE
Fix #18 PUM

### LINKED ISSUE #18: PUM
The Conditions Met Vector (CMV) can now be used in conjunction with the Logical Connector
Matrix (LCM) to form the Preliminary Unlocking Matrix (PUM). The entries in the LCM represent
the logical connectors to be used between pairs of LICs to determine the corresponding entry in
the PUM.

For example, LCM[i,j] represents the boolean operator to be applied to CMV[i] and CMV[j].
PUM[i,j] is set according to the result of this operation. If LCM[i,j] is NOTUSED, then PUM[i,j]
should be set to true. If LCM[i,j] is ANDD, PUM[i,j] should be set to true only if (CMV[i] AND
CMV[j]) is true. If LCM[i,j] is ORR, PUM[i,j] should be set to true if (CMV[i] OR CMV[j]) is
true. (Note that the LCM is symmetric, i.e. LCM[i,j]=LCM[j,i] for all i and j.)

### PR DESCRIPTION
**Pull request by sashahe**
Implement logic and add unit test for PUM.
Enums set to public to enable testing

### CODE REVIEW

File: src/test/java/DecideTest.java [hunk 1]
```diff
@@ -189,4 +189,52 @@ public void testLIC14() {
...
+
+    // Test a requirement where everything must be true
+    for (int i = 0; i < 15; i++) {
+      decide.CMV[i] = true;
+      for (int j =0; j < 15; j++) {
+        decide.LCM[i][j] = Decide.CONNECTORS.ANDD;
+      }
+    }
+
+    decide.calculatePUM();
```
Comment (EmelieTham) [inline src/test/java/DecideTest.java]: Hi! I might have misunderstood the code, but is there suppose to be a `assertTrue` for the test the requirement in which everything is true?

File: src/test/java/DecideTest.java [hunk 2]
```diff
@@ -342,4 +342,58 @@ public void testLIC14() {
...
+    Decide decide = new Decide();
+    decide.CMV[3] = true;
+
+    // Test a requirement where only LIC3 must be true
+    for (int i = 0; i < 15; i++) {
+      for (int j =0; j < 15; j++) {
+        if (i  == 3 || j == 3) {
+          decide.LCM[i][j] = Decide.CONNECTORS.ORR;
+        } else {
+         decide.LCM[i][j] = Decide.CONNECTORS.NOTUSED;
```
Comment (gustafguner) [inline src/test/java/DecideTest.java]: Formatting wise; add a space here 😈

File: src/test/java/DecideTest.java (hunk 1 above)
Comment (gustafguner) [reply src/test/java/DecideTest.java]: Isn't that the `assertTrue`below? Or is it @sashahe ?

Comment (sashahe) [reply src/test/java/DecideTest.java]: I just added it! It was not in the version @EmelieTham reviewed :)

File: src/test/java/DecideTest.java (hunk 2 above)
Comment (sashahe) [reply src/test/java/DecideTest.java]: Good catch! :+1:

File: src/test/java/DecideTest.java (hunk 1 above)
Comment (sashahe) [reply src/test/java/DecideTest.java]: I just added it but responded in the 'Files changed' view, and these comments do not seem to appear!

File: src/test/java/DecideTest.java (hunk 2 above)
Comment (sashahe) [reply src/test/java/DecideTest.java]: Nice catch :+1:

**Pull request merged**